### PR TITLE
adding devcontainer to open in Codespaces

### PR DESCRIPTION
**Pull request by blackgirlbytes**
## Description
This PR adds a devcontainer to open the TBD website in Codespaces. The dev container installs all necessary project dependencies, runs the project, and opens up a browser preview of the project. 

**Please note**: If you click the button before merging, it won't have any of the configurations I set up because it's not merged yet. You can try out this link to see what the experience would be like. This is the link that points to my personal branch: https://codespaces.new/galaxy-bytes/developer.tbd.website/tree/adding-codespaces-hacktoberfest?quickstart=1

The first time you click it, it might be slow, but it will be a little faster on additional attempts.

This option is for contributor use; Codesandbox will be for workshops/code snippets.

## What type of PR is this? (check all applicable)

- [x] 🆕  Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
N/A
## Mobile & Desktop Screenshots/Recordings
https://github.com/TBD54566975/developer.tbd.website/assets/22990146/06b74b51-95a8-4eac-8401-7f40b575f8f5

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [x] 📜 README.md
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?
N/A

### CODE REVIEW

**Comment by blackgirlbytes** [comment]
also side note: @angiejones @EbonyLouis -- what do yall think of this PR template? If you all haven't decided on one yet, we can use something like this for contributors (with changes that suit this organization)

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    "name": "TBD Website - GitHub Codespaces container",
```
Comment (angiejones) [inline .devcontainer/devcontainer.json:2]: TBDeveloper Website

**Review by angiejones — CHANGES_REQUESTED** [review]

**Comment by dayhaysoos** [comment]
Hey @blackgirlbytes, this looks really cool! I love that this is an easier way for contributors to jump right in.

I also really like the PR template, would be down to adapt it to our needs (things like web5 version etc etc).

Curious about what the others think

**Comment by EbonyLouis** [comment]
I like it, I had it as an item to do especially with hacktoberfest coming up, and yours is really cute! Nick and Andrew should be able to provide more guidance on the specific tweaks that are needed. 

I also love the option under test, "no, because I need help". Thanks Rizel!

**Comment by EbonyLouis** [comment]
@blackgirlbytes here are the other 3 repos that you can add codespaces too as well please

- `Dwn-server`
- `Dwn-sdk-js`
- `Web5-js`

**Review by angiejones — APPROVED** [review]

**Comment by blackgirlbytes** [comment]
Thanks before anyone merges this, let me know how yall feel about my Discord message

**Comment by blackgirlbytes** [comment]
Closing this PR as we're moving to code sandbox instead. Thank you!

**Pull request closed without merging**